### PR TITLE
Feat/pwa ify

### PR DESCRIPTION
**Pull request by thepassle**
NOTES:
- Ive been testing this on my own fork and on netlify. I had to remove the basepath `/lockdown` stuff to get things working. Im not sure how the rancher stuff works and where it will deploy to, but we should keep that in mind when we go to prod

- Pwa-asset-generator adds a bunch of weird whitespaces. This is because they use the Cheerio library, which is unmaintained. I asked the maintainer of pwa-asset-generator if he would consider minifying instead. The index.html still gets minified in the build though.

- Discuss what should be done with the mapbox tiles. Ux/dev think we could do without tiles, if we add country label names

TODO:

  - [x] ~~precache json data files?~~ runtime cache json data files Figure out which caching strategy to use. (networkfirst)
  - [x] test if there are any issues with basepath?
  - [ ] do we always want to host on github pages? it doesnt support spa routing/complicates the sw. Move to netlify?
  - [x] strategies for api calls/fetch calls
    - [x] api.coronatracker.com/
    - [x] travel-advisory.info/api
  - [x] strategies for dynamic imports
    - [x] leaflet is dynamically imported, this is added in the precache which is fine because it happens in the background
  - [x] strategy for https://api.mapbox.com/styles/v1/mapbox api requests
  - [x] add max entries to not hog the users device storage
  - [x] pwa-asset-generator doesnt minify html after adding to index.html, so we should run pwa-asset-generator on the src index.html instead
  - [x] add pwa-install-button component
  - [x] add pwa-update-available component
  - [x] handle offline state. If a response is in the cache, use it, if not, show user friendly offline message instead
  - [x] extensively test
  - [x] find solution for mapbox tiles. Maybe go without? Looks kinda nice actually. But then we need to add labels for countries somehow. Discuss with milo and ux group. We will go without for now. Can do this in a separate branch
  - [x] call reg.update() on spa navigations to check if there is a new sw available (otherwise if users keep the tab open forever they wont get updates)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,24 +1,306 @@
...
+
+</head>
   <body>
     <div id="app"></div>
     <script type="module" src="./src/app.js"></script>
-  </body>
-</html>
+
+
+</body></html>
```
Comment (LarsDenBakker) [inline index.html]: can we run the index.html through prettier after generating

File: src/app.js [hunk 2]
```diff
@@ -3,6 +3,13 @@ import { render } from 'preact';
 import { Header } from './components/Header.js';
 import { MainPage } from './pages/MainPage.js';
 
+const version = '1.0.2';
+console.log(version);
```
Comment (LarsDenBakker) [inline src/app.js]: We can make this a more meaningful message

File: src/app.js [hunk 3]
```diff
@@ -3,6 +3,13 @@ import { render } from 'preact';
 import { Header } from './components/Header.js';
 import { MainPage } from './pages/MainPage.js';
 
+const version = '1.0.2';
```
Comment (LarsDenBakker) [inline src/app.js]: We should automate the version in rollup, I can add this

File: src/components/Header.js [hunk 4]
```diff
@@ -56,7 +56,7 @@ const styles = css`
 
 export const Header = () => html`
   <header class=${styles}>
-    <a href="/lockdown/">
+    <a href="/">
```
Comment (LarsDenBakker) [inline src/components/Header.js]:
```suggestion
    <a href="">
```

This way it always takes the base href

File: index.html (hunk 1 above)
Comment (thepassle) [reply index.html]: How? 🙈 I tried adding html to the `format` script but it didnt seem to take the index.html, is there other config needed to run prettier on html files?

File: src/app.js (hunk 3 above)
Comment (thepassle) [reply src/app.js]: Made a separate issue for this: https://github.com/Code-for-All/lockdown/issues/82

File: index.html (hunk 1 above)
Comment (LarsDenBakker) [reply index.html]: We can look at it later.

**Comment by LarsDenBakker** [comment]
LGTM, should it still be WIP?

**Comment by thepassle** [comment]
No, this is good to go for now

**Pull request merged**